api.planx.uk/modules/analytics/metabase/shared/client.ts

```diff
@@ -0,0 +1,155 @@
+import axios from "axios";
+import type {
+  AxiosInstance,
+  AxiosError,
+  AxiosRequestConfig,
+  InternalAxiosRequestConfig,
+} from "axios";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+// Custom error class for Metabase-specific errors
+export class MetabaseError extends Error {
+  constructor(
+    message: string,
+    public statusCode?: number,
+    public response?: unknown,
+  ) {
+    super(message);
+    this.name = "MetabaseError";
+  }
+}
+
+interface MetabaseConfig {
+  baseURL: string;
+  apiKey: string;
+  timeout?: number;
+  retries?: number;
+}
+
+// Validate environment variables
+const validateConfig = (): MetabaseConfig => {
+  const baseURL = process.env.METABASE_BASE_URL;
+  const apiKey = process.env.METABASE_API_KEY;
+
+  if (!baseURL) {
+    throw new Error(
+      "METABASE_BASE_URL is not defined in environment variables",
+    );
+  }
+  if (!apiKey) {
+    throw new Error("METABASE_API_KEY is not defined in environment variables");
+  }
+
+  return {
+    baseURL,
+    apiKey,
+    timeout: parseInt(process.env.METABASE_TIMEOUT ?? "30000"), // 30 second default timeout
+    retries: parseInt(process.env.METABASE_MAX_RETRIES ?? "3"), // 3 default retries
```

Comment:
```suggestion
    timeout: METABASE_TIMEOUT,
    retries: METABASE_MAX_RETRIES,
```

I think these are really constants (`const METABASE_TIMEOUT = 30_000`) as opposed to env vars which can differ across environments.